src/index.ts

```diff
@@ -1,7 +1,17 @@
 export { SRPParameters, HashFunction, PrimeGroup } from "./parameters";
 export { SRPRoutines } from "./routines";
-export { SRPClientSession } from "./session-client";
-export { SRPServerSession } from "./session-server";
+export {
+  SRPClientSession,
+  SRPClientSessionStep1,
+  SRPClientSessionStep1State,
```

Comment:
Do we really need to export state classes? Not sure.

Reply:
First I prefer `type` over `interface`, contrary to TS general recommendation.

Second, I'd say we don't need to export these types, because between `toJSON` and `fromState` that types are definitely lost when the user stores it as strings. I think `toJSON` could just be the inferred type, and `fromState` could even use `state: unknown` to make it clear.

Reply:
I've switched to `type` and removed exporting state types. 

I disagree however, with the second part. I think having an explicit types for  toJSON and fromState makes sure the methods are compatible (toJSON's output is valid input for fromState).

Reply:
In practice it will pass through `JSON.stringify:string -> JSON.parse:any` though, which produces `any`. But the type info (on the method signature) makes it clear what is expected from the user, good.

Reply:
Hmm. If the user wanted to make a custom wrapper function that would then call .fromState, they have no ability to be typesafe as they can't access the state types. Maybe it's worth exporting them? 


```
class SomeWrapper {
   deserializeStep1(data: any){ // no type safety, i should be able to use SRPClientSessionStep1State here
      return SRPClientSessionStep1.fromState(this.routines, data);
   }
}
```

Reply:
Well, that `data:any` would come from `JSON.parse` or similar, no?

But if it irks you, go ahead and export the state types, it won't hurt either. It will probably increase readability, and let the user know what to expect, and that it's all JSON friendly structs.

Reply:
It does kinda irk me ;)

I'll add em back in.